### PR TITLE
bdfresize: init at 1.5

### PR DESCRIPTION
**Pull request by malte-v**
bdfresize is a little tool used to resize BDF bitmap fonts. The code is quite dated, so I had to patch out an unneeded declaration of `malloc`. Now it compiles and works fine.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/bdfresize/default.nix [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{ lib, stdenv }:
+
+stdenv.mkDerivation rec {
+  pname = "bdfresize";
+  version = "1.5";
```
Comment (fabaff) [inline pkgs/tools/misc/bdfresize/default.nix:5]: For 20 years there was no new release. I'm wondering if there are no alternatives around which are maintained.

Comment (malte-v) [reply pkgs/tools/misc/bdfresize/default.nix:5]: I don't know about any alternatives. There are probably GUI programs around that provide all the functionality of bdfresize, but I'd like to resize fonts programmatically from within a Nix builder. If you can point me to a maintained alternative then I would be happy to package that instead.

File: pkgs/tools/misc/bdfresize/default.nix [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    url = "http://openlab.ring.gr.jp/efont/dist/tools/bdfresize/${pname}-${version}.tar.gz";
+    sha256 = "1zccbv9nzs6344aq3kwlg91x1mxxnk5w0k9aa2amgggsi4j4p1hb";
+  };
+
+  patches = [ ./remove-malloc-declaration.patch ];
+
+  meta = with lib; {
+    description = "Tool to resize BDF fonts";
+    homepage = "http://openlab.ring.gr.jp/efont/dist/tools/bdfresize/";
+    license = licenses.gpl2;
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/bdfresize/default.nix:17]: gpl2Only or gpl2Plus?

File: maintainers/maintainer-list.nix [hunk 3]
```diff
@@ -5714,6 +5714,12 @@
     githubId = 115218;
     name = "Felix Richter";
   };
+  malvo = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:5717]: Please put this into a separate commit with the message ``maintainers: add malvo``.

**Comment by malte-v** [comment]
I don't need this anymore, but I also see no reason not to include it in nixpkgs. Yes, it isn't maintained, but it does its job and will likely still work for the next 20 years just like it did during the last 20 years without maintenance. For reference, Ubuntu, Debian, openSUSE and Slackware all have bdfresize in their main repositories.

**Comment by SuperSandro2000** [comment]
@malte-v please resolve the merge conflict.

**Comment by malte-v** [comment]
@SuperSandro2000 done, better late than never :-)

File: pkgs/tools/misc/bdfresize/default.nix [hunk 4]
```diff
@@ -0,0 +1,20 @@
+{ lib, stdenv }:
+
+stdenv.mkDerivation rec {
+  pname = "bdfresize";
+  version = "1.5";
+
+  src = fetchTarball {
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/bdfresize/default.nix:7]: Please use fetchurl or fetchzip.

**Comment by dtzWill** [comment]
Consider fetchpatch'ing the equivalent patch from Debian?
https://sources.debian.org/patches/bdfresize/1.5-12/

Haven't checked the man page, but that patch might be good to grab as well.

Iirc getting permalinks for these isn't always easy (or maybe that's not an issue anymore? IIRC it's due to the version being in the URL, I think they have another system that serves these more permanently...), so look how it's down elsewhere in-tree :).  While realistically it'll likely end up cached by us it's important that things stay live when possible.  I tried looking into this but on my phone and gah.  No worries.  Personally I'm fine either way, I kinda like just having small patches local but that's not generally preferred :innocent:.

Thanks for submitting this, curious what font/resizing you got in mind! Anyway, hope this helps!

Moving off fetchTarball is more important! :)

File: pkgs/tools/misc/bdfresize/default.nix (hunk 4 above)
Comment (malte-v) [reply pkgs/tools/misc/bdfresize/default.nix]: fixed

**Comment by malte-v** [comment]
@dtzWill i dunno, personally I'd prefer keeping the patch in-tree, too. My reasoning is that debian should not be considered an "upstream" for this package. The man page renders completely fine without any patching.
@SuperSandro2000 i think this is ready to merge now, if there are no further suggestions :)
